elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/BackendRequestProcessor.java

```diff
@@ -224,6 +239,11 @@ public void shutdown() {
 			scheduler.shutdown();
 			try {
 				scheduler.awaitTermination( Long.MAX_VALUE, TimeUnit.SECONDS );
+				if ( lastAsyncWorkLatch != null ) {
+					//It's possible that a task was successfully scheduled but had no chance to run,
```

Comment:
- you add some work, which schedules a job to run in X time. 
- you (or another thread) initiate shut down,  but X didn't start yet
- the executor is shutdown so the scheduked job is canceled
- other threads waiting for it are stuck forever